course/api/views.py

```diff
@@ -192,23 +191,30 @@ class CourseExercisesViewSet(NestedViewSetMixin,
     `GET /courses/<course_id>/exercises/<exercisemodule_id>/`:
         returns the details and exercises of a specific module.
     """
-    lookup_url_kwarg = 'exercisemodule_id'
-    lookup_value_regex = REGEX_INT
-    parent_lookup_map = {'course_id': 'course_instance.id'}
     serializer_class = CourseModuleSerializer
 
-    def get_queryset(self) -> QuerySet[CourseModule]:
-        return (
-            CourseModule.objects
-            .get_visible(self.request.user)
-            .prefetch_related(
-                Prefetch('learning_objects', BaseExercise.objects.all(), 'exercises')
-            )
-            .all()
+    #get all exercises of all modules from the cache
+    def list(self, request, *args, **kwargs):
+        serializer = self.serializer_class(
+            self.content.data['exercises'],
+            many=True,
+            context={ 'request': request }
         )
+        response_data = { 'results': serializer.data }
+        return Response(response_data)
 
-    def get_object(self):
-        return self.get_member_object('module', 'Exercise module')
+    #get the exercises of a specific course module (has O(n) complexity, should this scale better?)
+    def retrieve(self, request, version = None, course_id = None, pk=None):
+        for module in self.content.data['exercises']:
```

Comment:
Not ideal if we have to loop over the whole course contents in the cache just to find the given exercise or module...